models/skilledImpactRequestsModel.js

```diff
@@ -1,33 +1,73 @@
 const db = require('../database/dbConfig');
 
+const pick = require('../util/pick.js');
+
 async function findSkilledImpactRequests(campaign_id) {
   return db('project_goals')
     .where({ campaign_id })
     .fullOuterJoin('skilled_impact_requests', 'skilled_impact_requests.id', 'project_goals.skilled_impact_request_id')
     .select(
       '*',
     )
-    .then((returnedQuery) => {
-      const skilledRequests = new Map();
-      for (let i = 0; i < returnedQuery.length; i += 1) {
-        if (!skilledRequests.has(returnedQuery[i].id)) {
+    .then((rows) => {
+      const skilledRequestsMap = new Map();
+      for (let i = 0; i < rows.length; i += 1) {
+        const skilledRequest = rows[i];
+        if (!skilledRequestsMap.has(skilledRequest.id)) {
           const skillAndProject = {
-            skill: returnedQuery[i].skill,
-            point_of_contact: returnedQuery[i].point_of_contact,
-            welcome_message: returnedQuery[i].welcome_message,
-            our_contribution: returnedQuery[i].our_contribution,
+            skill: skilledRequest.skill,
+            point_of_contact: skilledRequest.point_of_contact,
+            welcome_message: skilledRequest.welcome_message,
+            our_contribution: skilledRequest.our_contribution,
             project_goals: [],
           };
-          skilledRequests.set(returnedQuery[i].id, skillAndProject);
+          skilledRequestsMap.set(skilledRequest.id, skillAndProject);
         }
         const projectGoal = {
-          goal_title: returnedQuery[i].goal_title,
-          description: returnedQuery[i].description,
+          goal_title: skilledRequest.goal_title,
+          description: skilledRequest.description,
         };
-        skilledRequests.get(returnedQuery[i].id).project_goals.push(projectGoal);
+        skilledRequestsMap.get(skilledRequest.id).project_goals.push(projectGoal);
       }
-      return Array.from(skilledRequests.values());
+      return Array.from(skilledRequestsMap.values());
     });
 }
 
-module.exports = { findSkilledImpactRequests };
+async function insertSkilledImpactRequests(skilledRequests, campaign_id) {
+  const skillProps = ['skill', 'point_of_contact', 'welcome_message', 'our_contribution'];
+  const projectGoals2DArr = [];
+  const skilledRequestArr = skilledRequests.map((skilledRequest) => {
+    const skilledRequestWithCampaignId = {
+      ...pick(skilledRequest, skillProps),
+      campaign_id,
+    };
+    projectGoals2DArr.push(skilledRequest.project_goals);
```

Comment:
Since you're using `map` (a functional programming concept), this array operation is a good example of what's called a "side effect" in FP (see the "Side Effects" section [here](https://medium.com/javascript-scene/master-the-javascript-interview-what-is-functional-programming-7f218c68b3a0)).